### DISCUSSION TITLE
Updating syntax definitions

### DISCUSSION THREAD

**Discussion by stefanobaghino** [dandavison/delta#1702] [Ideas]
👋🏻 Hello, everyone!

A few weeks ago, I noticed that syntax highlighting didn't pick up Java text blocks (i.e., multiline strings). I started looking at the problem, which led me to learn that `delta` uses `syntect` for syntax highlighting, the same library `bat` uses.

Some more investigation made me realize that `bat` itself is lagging behind because ultimately `syntect` itself is. `bat` hasn't bumped its themes in a few years because of https://github.com/trishume/syntect/issues/323 and https://github.com/trishume/syntect/issues/271, which prevent the adoption of more modern themes.

I want to help, but first, I'd like to understand the maintainer's point of view on this matter so that my contribution can actually be helpful: is syntax highlighting compatibility with `bat` a core feature of `delta` (the `README` seem to point in that direction), or would you consider the idea of adopting a different syntax highlighting library? Or do you have a completely different approach in mind?

I am looking forward to your feedback; all the best!

**Comment by dandavison** [comment]
Hi @stefanobaghino! First I should be clear that my current job is providing an unlimited amount of interesting software challenges and I do not currently have time to take on a substantial delta project such as swapping out the syntax highlighter myself.

Beyond that, my initial thoughts are:
- In addition to the initial implementation, maintaining a high quality library of language syntax and theme definitions is non-trivial work. Is there a project you have in mind that would do that for delta? It has been extremely helpful to the delta project that bat has maintained these. (Thanks bat team!)
- What's good for delta would presumably be good for bat too: we have the same requirements here AFAIK. So if the bat team are _not_ taking this step, we'd probably want to understand why.
- My impression is that two of the bat maintainers/contributors have taken over syntect maintenance (@Enselic and @keith-hall). They must have thought quite a lot about this issue.

**Reply by keith-hall** [reply]
We have had the idea of implementing support for other syntax highlighting engines like tree-sitter  in `bat`, but nothing has been decided yet, and as some contributors have stepped forward to improve `syntect` (https://github.com/trishume/syntect/pull/536), likely that gives us reason to keep using it.

**Reply by stefanobaghino** [reply]
@dandavison Thanks for sharing your thoughts and for involving @keith-hall!

@keith-hall I'll move to bat's discussion board to avoid spamming here. 🙇🏻
